### DISCUSSION TITLE
Why does large detection performance decrease when image size is increased?

### DISCUSSION THREAD

**Discussion by YoungjaeDev** [obss/sahi#650] [Q&A] [answered]
Hi! 
Thank you for your contribution first 
The test image size is 1280x720 video When I first made standard prediction 640, sliding 640x640, the objects that were close to me were well caught, but the things that were far away were appear fluctation
So I did standard prediction 1280, sliding 640x640, but why can not I catch the things that are close to me when I catch the objects that are close and the things that are far away are quite catchy?

**Comment by fcakyon** [comment]
@youngjae-avikus can please explain your issue in detail? 
Are you performing standard inference or sliced inference or both? 
What is the `slice_height/width` and `input_size`? 
What is the `model_type`, yolov5/detectron/mmdet ?

When `input_size=1280` and `slice_size=640`, `sahi` will enhance each 640 slice into 1280 (2x enhancing) which makes larger objects huge and thus the large object detection performance may decrease. You have to arrange these parameters to fit your need.

**Comment by YoungjaeDev** [comment]
@fcakyon 

**Rather, the large object image is 640->1280 and becomes bigger when it is resized, so the performance is adversely affected, right?**
I haven't done a lot of experiments yet, so I'm going to do an experiment and ask you a question

eg. **if separate the two models and give them different image sizes**, wouldn't the above issues be eased a little?

```python
standard_detection_model = Yolov5DetectionModel(
    model_path=weights,
    confidence_threshold=conf_thres,
    device=device, # or 'cuda:0'
    image_size=640
)

slice_detection_model = Yolov5DetectionModel(
    model_path=slice_weights,
    confidence_threshold=conf_thres,
    device=device, # or 'cuda:0'
    image_size=320
)
```

Thank you!

**Comment by fcakyon** [comment, marked as answer]
> eg. **if separate the two models and give them different image sizes**, wouldn't the above issues be eased a little?
> 
> ```python
> standard_detection_model = Yolov5DetectionModel(
>     model_path=weights,
>     confidence_threshold=conf_thres,
>     device=device, # or 'cuda:0'
>     image_size=640
> )
> 
> slice_detection_model = Yolov5DetectionModel(
>     model_path=slice_weights,
>     confidence_threshold=conf_thres,
>     device=device, # or 'cuda:0'
>     image_size=320
> )
> ```

@youngjae-avikus you are definitely right on that! See this reply for a related discussion: https://github.com/obss/sahi/issues/302#issuecomment-1026812923
